src/Microsoft.SqlTools.Credentials/Credentials/OSX/OSXCredentialStore.cs

```diff
@@ -45,14 +46,42 @@ public bool Save(Credential credential)
             return result;
         }
 
+        public bool GetCredentials(string credentialPattern, out List<Credential> credentials)
+        {
+            var query = new List<(string, string)>
+            {
+                ("kSecClass", "kSecClassGenericPassword"),
+                ("kSecAttrService", credentialPattern),
```

Comment:
So what is the pattern you're planning on using here? Let's be careful not to accidently be deleting other credentials that don't belong to us. Maybe it'd be enough to list the credentials to delete for the user to accept before doing the actual delete?

Reply:
Nothing in SqlToolsService, here it will support any generic pattern.